CrewChiefV4/Events/Fuel.cs

```diff
@@ -1209,7 +1224,35 @@ public int getLitresToEndOfRace()
             }
             return additionalLitresNeeded;
         }
-
+        // Try to predict the the earliest possible time/lap and the latest possible time/lap we can come in for our pitstop and still make it to the end.
+        // we need to check if more then one stop is needed to finish the race in this case we dont care about pit window
+        public Tuple<int, int> getPredictedPitWindow()
+        {
+            Tuple<int, int> pitWindow = new Tuple<int, int>(-1, -1);
+            if (lapsCompletedSinceFuelReset > 3 && averageUsagePerLap > 0)
+            {
+                int litersNeeded = getLitresToEndOfRace();
+                gotPredictedPitWindow = true;
+                if (litersNeeded > 0)
+                {
+                    // more then 1 stop needed
+                    if (litersNeeded > fuelCapacity)
+                    {
+                        return pitWindow;
+                    }
+                    int maximumLapsForFullTankOfFuel = (int)Math.Floor(fuelCapacity / averageUsagePerLap);
+                    int pitWindowEnd = (int)Math.Floor(initialFuelLevel / averageUsagePerLap); //pitwindow end
+                    int estimatedlapsWorth = (int)Math.Floor(litersNeeded / averageUsagePerLap);
+                    int diff = maximumLapsForFullTankOfFuel - pitWindowEnd;
+                    int pitWindowStart = (maximumLapsForFullTankOfFuel - diff) - estimatedlapsWorth;
+
+                    Console.WriteLine("pitwindowStart: " + pitWindowStart + " pitWindowEnd: " + pitWindowEnd +
+                        "maximumLapsForFullTankOfFuel: " + maximumLapsForFullTankOfFuel + " estimatedlapsWorth: " + estimatedlapsWorth);
```

Comment:
I'd suggest prepending this message with "Predicted Pit Window: " otherwise it seems without context.

Reply:
It already does that, on line 667 it plays the message. but it needs another estimate recording as the one used is only good for fuel.